src/ukify/ukify.py

```diff
@@ -858,6 +893,55 @@ def generate_keys(opts):
             pub_key.write_bytes(pub_key_pem)
 
 
+def pe_read_single_section(section, opts, name, text):
+    data = section.get_data(ignore_padding=True)
+    size = section.Misc_VirtualSize
+    h = sha256(data).hexdigest()
+    data_printed = None
+
+    tmp = {
+        'size' : size,
+        'sha256' : h,
+    }
+
+    if text:
+        try:
+            data_printed = data.decode()
+        except UnicodeDecodeError:
+            print(f"Warning! {name} cannot be read. Not a valid UTF8-encoded text")
+            data_printed = None
+        finally:
+            tmp['text'] = data_printed
+
+    if opts.json:
+        out_str = json.dumps(tmp)
+    else:
+        out_str = f"{name}:\nsize: {size} bytes\nsha256: {h}\n"
+        if text:
+            out_str += f"text:\n{data_printed}\n"
+        print(out_str)
+
+    if name in Section.custom_out_file:
+        Section.custom_out_file[name].write_text(out_str)
+
+    return tmp
+
+
+def pe_read_sections(opts):
```

Comment:
The name is misleading. This "job" of this function is to prints sections. The fact that it reads them is just a step towards that goal.